components/delete_post_modal/delete_post_modal.jsx

```diff
@@ -1,106 +1,95 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See License.txt for license information.
 
+import PropTypes from 'prop-types';
 import React from 'react';
 import {Modal} from 'react-bootstrap';
 import {FormattedMessage} from 'react-intl';
 
-import {deletePost} from 'actions/post_actions.jsx';
-import ModalStore from 'stores/modal_store.jsx';
-import Constants from 'utils/constants.jsx';
 import * as UserAgent from 'utils/user_agent.jsx';
 
-var ActionTypes = Constants.ActionTypes;
+export default class DeletePostModal extends React.PureComponent {
+    static propTypes = {
 
-export default class DeletePostModal extends React.Component {
-    constructor(props) {
-        super(props);
+        /**
+         * post data
+         */
+        post: PropTypes.object.isRequired,
 
-        this.state = {
-            show: false,
-            post: null,
-            commentCount: 0,
-            error: '',
-        };
-    }
+        /**
+         * post comment count
+         */
+        commentCount: PropTypes.number.isRequired,
 
-    componentDidMount() {
-        ModalStore.addModalListener(ActionTypes.TOGGLE_DELETE_POST_MODAL, this.handleToggle);
-    }
+        /**
+         * Does the post come from RHS mode
+         */
+        isRHS: PropTypes.bool.isRequired,
 
-    componentWillUnmount() {
-        ModalStore.removeModalListener(ActionTypes.TOGGLE_DELETE_POST_MODAL, this.handleToggle);
-    }
+        /**
+        * Function called when modal is dismissed
+        */
+        onHide: PropTypes.func.isRequired,
 
-    componentDidUpdate(prevProps, prevState) {
-        if (this.state.show && !prevState.show) {
-            setTimeout(() => {
-                if (this.deletePostBtn) {
-                    this.deletePostBtn.focus();
-                }
-            }, 200);
-        }
+        actions: PropTypes.shape({
+
+            /**
+            * Function called for deleting post,
+            */
+
+            deletePost: PropTypes.func.isRequired,
+        }),
     }
 
-    handleDelete = () => {
-        deletePost(
-            this.state.post.channel_id,
-            this.state.post,
-            () => {
-                this.handleHide();
-            },
-            (err) => {
-                this.setState({error: err.message});
-            }
-        );
+    constructor(props) {
+        super(props);
+        this.handleDelete = this.handleDelete.bind(this);
+        this.onHide = this.onHide.bind(this);
```

Comment:
No action required but just a heads up if you use arrow syntax you don't need these binds (since it will autobind). e.g

```
handleDelete = () => {
// stuff
}
```

Reply:
Oh, that is good to know. Thanks!